src/pages/OfferNew/Introduction/Sidebar/DetailsModal/index.tsx

```diff
@@ -0,0 +1,425 @@
+
+  @media (max-width: 800px) {
+    width: 100%;
+    margin: 0 0 2rem 0;
+
+    :last-child {
+      margin-bottom: 0;
+    }
+  }
+`
+
+const ContentColumnTitle = styled.div`
+  font-size: 1rem;
+  color: ${colorsV2.gray};
+  margin-bottom: 1rem;
+  padding-left: 0.5rem;
+  display: flex;
+  flex-direction: row;
+  justify-content: space-between;
+  align-items: center;
+  min-height: 2rem;
+
+  @media (max-width: 800px) {
+    padding: 0;
+  }
+`
+
+const ContentColumnTitleButton = styled.button`
+  padding: 0.375rem 1rem;
+  font-size: 0.875rem;
+  color: ${colorsV2.gray};
+  border-radius: 24px;
+  border: 1px solid ${colorsV2.gray};
+  cursor: pointer;
+  transition: all 250ms;
+  background: none;
+
+  :focus {
+    outline: none;
+  }
+
+  :hover {
+    color: ${colorsV2.darkgray};
+    border-color: ${colorsV2.darkgray};
+  }
+`
+
+const Footer = styled.div`
+  margin-top: 2.5rem;
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  justify-content: center;
+`
+
+const Warning = styled.div`
+  font-size: 0.75rem;
+  line-height: 1rem;
+  margin-top: 1rem;
+  color: ${colorsV2.darkgray};
+  text-align: center;
+`
+
+const Error = styled.div`
+  font-size: 0.75rem;
+  line-height: 1rem;
+  margin-top: 1rem;
+  color: ${colorsV2.coral500};
+  text-align: center;
+`
+
+const LoadingDimmer = styled.div<{ visible: boolean }>`
+  background: rgba(255, 255, 255, 0.5);
+  position: absolute;
+  top: 0;
+  left: 0;
+  right: 0;
+  bottom: 0;
+  transition: all 350ms;
+  opacity: ${(props) => (props.visible ? 1 : 0)};
+  visibility: ${(props) => (props.visible ? 'visible' : 'hidden')};
+`
+
+interface DetailsModalProps {
+  quote: CompleteQuote
+  refetch: () => void
+}
+
+export const DetailsModal: React.FC<ModalProps & DetailsModalProps> = ({
+  quote,
+  refetch,
+  isVisible,
+  onClose,
+}) => {
+  const textKeys = useTextKeys()
+  const [editQuote, editQuoteResult] = useEditQuoteMutation()
+  const fieldSchema = getFieldSchema(quote)
+  const validationSchema = getValidationSchema(fieldSchema, quote)
+  const initialValues = getInitialValues(quote)
+  const [isUpdating, setIsUpdating] = React.useState(false)
+
+  return (
+    <Modal isVisible={isVisible} onClose={onClose} dynamicHeight>
+      <LoadingDimmer visible={isUpdating} />
+      <Container>
+        <Formik<EditQuoteInput>
+          initialValues={initialValues}
+          validationSchema={validationSchema}
+          validateOnBlur
+          onSubmit={(form) => {
+            setIsUpdating(true)
+
+            editQuote({ variables: { input: form } })
+              .then(async (result) => {
+                if (!result || (result.errors && result.errors.length > 0)) {
+                  setIsUpdating(false)
+                  return
+                }
+
+                await refetch()
+                onClose()
+                setIsUpdating(false)
+              })
+              .catch(() => {
+                setIsUpdating(false)
+              })
```

Comment:
i think we should pick either async/await or `.then` and not mix them, imo that becomes a little messy.

Reply:
Very true, nice input